src/Payments.sol

```diff
@@ -296,24 +424,47 @@ contract Payments is
         Account storage payer = accounts[rail.token][rail.from];
         Account storage payee = accounts[rail.token][rail.to];
 
+        uint256 oldRate = rail.paymentRate;
+
+        // For terminated rails, can only reduce rate, not increase it
+        if (rail.terminationEpoch > 0) {
+            require(
+                newRate <= oldRate,
+                "failed because terminated at epoch: cannot increase rate on terminated rail"
+            );
+            require(
+                oneTimePayment <= rail.lockupFixed,
+                "failed because terminated at epoch: one-time payment exceeds fixed lockup"
+            );
+        }
+
         OperatorApproval storage approval = operatorApprovals[rail.token][
             rail.from
         ][rail.operator];
-        require(approval.isApproved, "Operator not approved");
 
-        // Settle the payer's lockup.
-        // If we're changing the rate, we require full settlement.
+        // Settle the payer's lockup to account for elapsed time
         (bool fullySettled, uint256 lockupSettledUpto) = settleAccountLockup(
             payer
         );
-        if (newRate != rail.paymentRate) {
-            require(
-                fullySettled && lockupSettledUpto == block.number,
-                "account lockup not fully settled; cannot change rate"
-            );
-        }
 
-        uint256 oldRate = rail.paymentRate;
+        // If we're increasing the rate, we require full account lockup setlement before we can proceed
+        require(
+            newRate <= oldRate ||
+                (fullySettled && lockupSettledUpto == block.number),
+            "account lockup not fully settled; cannot increase rate"
+        );
+
+        // We just settled the account lockup. If the client is still not able to guaruantee
```

Comment:
I'd find this easier to understand if we broke it down case by case as separate conditionals:

1. Case 1: lockup not fully settled.
    1. Case 1a: in debt: require that rate must not change.
    2. Case 1b: not in debt: require that rate is less than or equal to the current rate.
3. Case 2: lockup fully settled, nothing to check.